src/main/java/com/iopipe/plugin/eventinfo/EventInfoDecoders.java

```diff
@@ -100,31 +107,52 @@ public EventInfoDecoders()
 		if (__o == null)
 			return new CustomMetric[0];
 		
+		EventInfoDecoder decoder = null;
+		
+		// The class type of the input object
 		Class<?> oftype = __o.getClass();
 		
-		// Go through
-		Class<?> match = null;
-		for (Class<?> maybe : this._classes)
+		// Check to see if the last decoder used is the one we want
+		EventInfoDecoder last = this._last;
+		if (last != null)
```

Comment:
This section below could possibly be moved to a method like `getDecoder` or some ilk, might make this easier to read (where the method returns the decoder to be used)